Waterfall.cpp

```diff
@@ -805,8 +807,27 @@ void Waterfall::mousePressEvent(QMouseEvent * event)
             {
                 if (m_BookmarkTags[i].first.contains(event->pos()))
                 {
-                    m_DemodCenterFreq = m_BookmarkTags[i].second;
+                    BookmarkInfo info = m_BookmarkTags[i].second;
+
+                    qInfo() << "bookmark '" << info.name
+                        << " ', freq: " << info.frequency
+                        << ", low: " << info.lowFreqCut
+                        << ", high: " << info.highFreqCut
+                        << ", modulation: " << info.modulation;
+
+                    if(!info.modulation.isEmpty()) {
+                        emit newModulation(info.modulation);
+                    }
+
+                    m_DemodCenterFreq = info.frequency;
                     emit newDemodFreq(m_DemodCenterFreq, m_DemodCenterFreq - m_CenterFreq);
+
+                    if(info.lowFreqCut != 0 && info.highFreqCut != 0) {
```

Comment:
What happens with SSB modulations? In USB, info.highFreqCut = 0 and in LSB, info.lowFreqCut = 0

Reply:
changed to bandwidth(), thanks :)